### PR TITLE
Add SimpleReportConfiguration API endpoint

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Spec [here](https://docs.google.com/document/d/1qkwGcl4GmVYYBnf5jzxOgfS_FE7kE6xlyAhvcSAtP-I/edit#heading=h.blt0fbwgkgt2).

This API endpoint returns a very simplified view of the `ReportConfiguration` model.
@gcapalbo

### CODE REVIEW

File: corehq/apps/api/tests.py [hunk 1]
```diff
@@ -1514,3 +1516,91 @@ def test_wrong_user_api_key(self):
...
+        cls.report_configuration = ReportConfiguration(
+            domain=cls.domain.name,
+            config_id=cls.data_source._id,
+            columns=cls.report_columns,
+            filters=cls.report_filters
+        )
+        cls.report_configuration.save()
+
+
+    def test_get_detail(self):
```
Comment (dimagimon) [inline corehq/apps/api/tests.py]: E303 too many blank lines (2)

**Comment by gcapalbo** [comment]
:+1: looks good to me. where does the auth happen, in one of the superclasses of the SimpleReportConfigurationResource? it might be worth adding a few lines to the test case to test accessing the resource with invalid credentials to make sure it rejects the request

File: corehq/apps/api/resources/v0_5.py [hunk 2]
```diff
@@ -489,3 +495,30 @@ def dehydrate(self, bundle):
         bundle.data['product_name'] = bundle.obj.sql_product.name
         bundle.data['transaction_date'] = bundle.obj.report.date
         return bundle
+
+
+class SimpleReportConfigurationResource(CouchResourceMixin, HqBaseResource, DomainSpecificResourceMixin):
```
Comment (gcapalbo) [inline corehq/apps/api/resources/v0_5.py]: i see the spec also mentions exposing an endpoint that runs the ucr report - is that here too? it seemed like this was just listing the configuration of the report unless i'm missing something

Comment (NoahCarnahan) [inline corehq/apps/api/resources/v0_5.py]: Nope, that will come in a future PR. Sorry for the confusion.

**Comment by NoahCarnahan** [comment]
The auth happens in the meta classes parent, `CustomResourceMeta`. Adding a test case sounds like a good idea though, thanks.

**Comment by gcapalbo** [comment]
:+1: on :white_check_mark:

**Comment by NoahCarnahan** [comment]
✅

**Pull request merged**